### ISSUE TITLE
Test suite has a build failure

### ISSUE THREAD

**Issue by mightybyte** [emilypi/Base16#9]
In some configurations, the test suite build fails with the following:

```
Preprocessing test suite 'tasty' for base16-0.3.0.1..
Building test suite 'tasty' for base16-0.3.0.1..
[1 of 2] Compiling Internal         ( test/Internal.hs, dist/build/tasty/tasty-tmp/Internal.o )
[11 of 21] Compiling Data.ByteString.Lazy.Base64 ( src/Data/ByteString/Lazy/Base64.hs, dist/build/Data/ByteString/Lazy/Base64.o, dist/build/Data/ByteString/Lazy/Base64.dyn_o )
[12 of 21] Compiling Data.ByteString.Lazy.Base64.URL ( src/Data/ByteString/Lazy/Base64/URL.hs, dist/build/Data/ByteString/Lazy/Base64/URL.o, dist/build/Data/ByteString/Lazy/Base64/URL.dyn_o )
[2 of 2] Compiling Main             ( test/Main.hs, dist/build/tasty/tasty-tmp/Main.o )

test/Main.hs:169:36: error:
    • Couldn't match expected type ‘(BS.ByteString, b0)’
                  with actual type ‘Either String BS.ByteString’
    • In the second argument of ‘($)’, namely
        ‘Bos.decode $ Bos.encode bs’
      In the first argument of ‘Right’, namely
        ‘(fst $ Bos.decode $ Bos.encode bs)’
      In the second argument of ‘(==)’, namely
        ‘(Right (fst $ Bos.decode $ Bos.encode bs) ::
            Either T.Text BS.ByteString)’
```

**Comment by sternenseemann** [comment]
Needs either a constraint on `base16-bytestring` (`< 1.0.0.0` should work) or preferrably an adjustment in the code which should be easy.

**Comment by emilypi** [comment]
Fixed via #10

**Comment by sternenseemann** [comment]
Thanks!
